trunk/SUAVE/Input_Output/OpenVSP/vsp_wing.py

```diff
@@ -299,7 +302,31 @@ def read_vsp_wing(wing_id, units_type='SI',write_airfoil_file=True):
     wing.twists.root      = vsp.GetParmVal(wing_id, 'Twist', 'XSec_0') * Units.deg
     wing.twists.tip       = vsp.GetParmVal(wing_id, 'Twist', 'XSec_' + str(segment_num-1)) * Units.deg
 
-
+    # check if control surface (sub surfaces) are defined
+    num_cs = vsp.GetNumSubSurf(wing_id)
+    for cs_idx in range(num_cs):
+        cs_id   = vsp.GetSubSurf(wing_id,cs_idx)
+        param_names = vsp.GetSubSurfParmIDs(cs_id)
+        for p_idx in range(len(param_names)):
+            if 'LE_Flag' == vsp.GetParmName(param_names[p_idx]):
+                LE_flag  = vsp.GetParmVal(param_names[p_idx])
+            if 'UStart' == vsp.GetParmName(param_names[p_idx]):
+                span_fraction_start   = vsp.GetParmVal(param_names[p_idx])
+            if 'UEnd' == vsp.GetParmName(param_names[p_idx]):
+                span_fraction_end     = vsp.GetParmVal(param_names[p_idx])
+            if 'Length_C_Start' == vsp.GetParmName(param_names[p_idx]):
+                chord_fraction        = vsp.GetParmVal(param_names[p_idx])
+        if LE_flag == 1.0:
+            CS = SUAVE.Components.Wings.Control_Surfaces.Slat()
+        else:
+            CS = SUAVE.Components.Wings.Control_Surfaces.Flap()
```

Comment:
It doesn't necessarily make it a flap. The rotation styles in the VLM differ between flaps and ailerons.

Can you check a couple things? First, if it's vertical, make it a rudder. Second, the most outboard control surface should be an aileron. All others can be flaps. I'm not sure how to check for horizontal stabilizers.

Reply:
Thanks for the suggestions. Updates made